Assets/Scripts/ConnectionManagement/ConnectionState/ClientConnectedState.cs

```diff
@@ -27,7 +26,8 @@ public override void Enter()
         public override void OnClientDisconnect(ulong _)
         {
             var disconnectReason = m_ConnectionManager.NetworkManager.DisconnectReason;
-            if (string.IsNullOrEmpty(disconnectReason))
+            if (string.IsNullOrEmpty(disconnectReason) ||
+                disconnectReason == "Disconnected due to host shutting down.")
```

Comment:
@ShadauxCat I'd like to get your eyes on this section of this PR that updates to new RPCs. For context I had to update this part of the connection management system because one of our tests was now failing.

I noticed that this [PR](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/commit/654e4134f7932644c4e34855e090f6703eb9a966) introduced a shutdown disconnect reason message delivered to connected clients when server/host shuts down. We assumed that when a host shut down, the DisconnectReason field would be unset, hence why our tests failed.

I'm curious if we should convert this string into an enum (or something along those lines) that we can compare against in game code. This line works for now, but this string may change in the future.

Let me know what you think!